src/python/WMCore/ACDC/DataCollectionService.py

```diff
@@ -91,10 +91,16 @@ def failedJobs(self, failedJobs, useMask=True):
             fileset = CouchFileset(database=self.database, url=self.url,
                                    name=taskName)
             coll.addFileset(fileset)
+            inputFiles = job['input_files']
+            for fInfo in inputFiles:
+                if fInfo["merged"] and ("parents" in fInfo) and \
```

Comment:
Same comment as below. I think `parents` key is always present, no? Just to be safe, it could be replaced by
```
if fInfo["merged"] and len(fInfo.get("parents", []) and fInfo["parents"].startswith("/store/unmerged"):
```

Now thinking about it in plain english: if the input file is merged and there are unmerged parents, skip the parents. I wonder whether we should skip any parents that are unmerged?

Reply:
I think parents key should be always present but I just checked for the safe side. I will change that.

Reply:
I think there shouldn't be a case if there is unmerged input and parents are unmerged.
It there are I think we have a bigger problem (parentage will be lost) I just want to record this in ACDC if that happens.

Reply:
Seangchan, as we discussed yesterday, our only concern is related to the unmerged parents file.
How about we remove the "merged" part in order to make it more clear.
BTW, do you know whether retrieving the first element of that iterator is faster than a simple `fInfo["parents"][0]`? Otherwise I'd just stick to the direct access to the first element.

Reply:
As we chat earlier, I think we should just check whether input file is merged file or not. We don't have to check about parents' file status (merged or unmered) but for the safe side, we just check parent file status. If you think it is safe to just check merged == 1 or not, we can change it later.